### PR TITLE
Main

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: src/main/java/core/mate/academy/model/Bulldozer.java [hunk 1]
```diff
@@ -4,10 +4,34 @@
  * Add some custom fields that could be only in Bulldozer
  * Do not remove no-args constructor
  */
-public class Bulldozer extends Machine {
+public class Bulldozer extends Machine implements Workable {
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/Bulldozer.java:7]:
```suggestion
public class Bulldozer extends Machine {
```

File: src/main/java/core/mate/academy/model/Excavator.java [hunk 2]
```diff
@@ -4,10 +4,34 @@
  * Add some custom fields that could be only in Excavator
  * Do not remove no-args constructor
  */
-public class Excavator extends Machine {
+public class Excavator extends Machine implements Workable {
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/Excavator.java:7]:
```suggestion
public class Excavator extends Machine {
```

File: src/main/java/core/mate/academy/model/Truck.java [hunk 3]
```diff
@@ -4,10 +4,34 @@
  * Add some custom fields that could be only in Truck
  * Do not remove no-field constructor
  */
-public class Truck extends Machine {
+public class Truck extends Machine implements Workable {
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/model/Truck.java:7]:
```suggestion
public class Truck extends Machine {
```

File: src/main/java/core/mate/academy/model/Bulldozer.java (hunk 1 above)
Comment (kozhukhovsky) [reply src/main/java/core/mate/academy/model/Bulldozer.java:7]: It's too much. Machine already implements this interface, so all inheritors will also implement this interface.

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:8]:
```suggestion

```

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    @Override
+    public List<Bulldozer> get() {
+        Bulldozer bulldozer1 = new Bulldozer(300, "Heavy");
+        Bulldozer bulldozer2 = new Bulldozer(120, "Light");
+        Bulldozer bulldozer3 = new Bulldozer(200, "Digging");
+        List<Bulldozer> bulldozers = new ArrayList<>();
+        bulldozers.add(bulldozer1);
+        bulldozers.add(bulldozer2);
+        bulldozers.add(bulldozer3);
+        return bulldozers;
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:18]: U can do it in loop

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+
+    @Override
+    public List<Bulldozer> get() {
+        Bulldozer bulldozer1 = new Bulldozer(300, "Heavy");
+        Bulldozer bulldozer2 = new Bulldozer(120, "Light");
+        Bulldozer bulldozer3 = new Bulldozer(200, "Digging");
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:13]:
```suggestion
        Bulldozer bulldozer1 = new Bulldozer(300, "Heavy");
        Bulldozer bulldozer2 = new Bulldozer(120, "Light");
        Bulldozer bulldozer3 = new Bulldozer(200, "Digging");

```

It's bad naming style when we use numbers.

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    @Override
+    public List<Excavator> get() {
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:9]: same as buldozer

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/TruckProducer.java:8]: remove empty line

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 9]
```diff
@@ -26,13 +28,13 @@ public interface MachineService {
      * @param machines - list of machines to be filled with value
      * @param value    - any object of machine sub class
      */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super Machine> machines, Machine value);
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineService.java:31]:
```suggestion
    void fill(List<? super T> machines, T value);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 10]
```diff
@@ -26,13 +28,13 @@ public interface MachineService {
...
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```
Comment (kozhukhovsky) [inline src/main/java/core/mate/academy/service/MachineService.java:39]:
```suggestion
    void startWorking(List<? extends T> machines);
```

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/BulldozerProducer.java (hunk 5 above)
Comment (OlegMR0) [reply src/main/java/core/mate/academy/service/BulldozerProducer.java:18]: Actually can't understand what you mean, all bulldozer variables are different.

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**